### DISCUSSION TITLE
How to perform action after rendering has been done?

### DISCUSSION THREAD

**Discussion by shi-yan** [solidjs/solid#1990] [Q&A] [answered]
in VUE, for example, there is a $nextTick function, in its callback, we can be sure the current around of rendering has been done.

in my solidjs code, I have an reactive array

```javascript
  const [items, setItems] = createSignal([]);
```

after I update the item array, the ui will be updated. Then after the ui has been updated, I want to scroll its container div to the bottom:

but I don't know how to make sure ui update has been done? i.e. what's solidjs equivalent of $nextTick?

right now, I'm doing it using setTimeout, which is stupid:

```javascript
        setItems([...items(), 28]);
        setTimeout(() => { containerDiv.scrollTop = containerDiv.scrollHeight; }, 1000);

```

in VUE I would write:

```javascript
        setItems([...items(), 28]);

        this.$nextTick(() => {
                  containerDiv.scrollTop = containerDiv.scrollHeight;
        })

```

**Comment by shi-yan** [comment, marked as answer]
I found it

https://www.solidjs.com/tutorial/introduction_effects

**Reply by ppzreboot** [reply]
But, How?
